airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -130,17 +130,21 @@ public AirbyteCatalog discover(final JsonNode config) throws Exception {
 
   @Override
   public List<TableInfo<CommonField<JDBCType>>> discoverInternal(JdbcDatabase database) throws Exception {
-    List<TableInfo<CommonField<JDBCType>>> internals = new ArrayList<>();
-    for (String schema : schemas) {
-      LOGGER.debug("Discovering schema: {}", schema);
-      internals.addAll(super.discoverInternal(database, schema));
-    }
-
-    for (TableInfo<CommonField<JDBCType>> info : internals) {
-      LOGGER.debug("Found table: {}", info.getName());
+    if (schemas != null && !schemas.isEmpty()) {
+      // process explicitly selected (from UI) schemas
+      final List<TableInfo<CommonField<JDBCType>>> internals = new ArrayList<>();
+      for (String schema : schemas) {
+        LOGGER.debug("Discovering schema: {}", schema);
+        internals.addAll(super.discoverInternal(database, schema));
+      }
+      for (TableInfo<CommonField<JDBCType>> info : internals) {
+        LOGGER.debug("Found table: {}", info.getName());
```

Comment:
Nitpick. Some tables may have the same names, but under different schemas. If we are going to log them, I think it's better to log the schema as well:

```suggestion
        LOGGER.debug("Found table (schema: {}): {}", info.getNameSpace(), info.getName());
```

Reply:
Updated, thanks